Automattic/jetpack
# Refactor: Decouple Social's ToggleSection component

This change decouples the `ToggleSection` component, so it's not tied to the logic behind it.

## Proposed changes:

* Decoupled the `ToggleSection` component, now it has props and handlers.
* Added some type definitions to support our motion to use more TypeScript with our components. p1678866772214549-slack-C02JJ910CNL
* Created a new `SocialModuleToggle` component that utilizes the decoupled `ToggleSection` component
* Removed the `ModuleToggle` as after decoupling it was only some styling to the `ToggleControl` component

### Other information:


## Jetpack product discussion



1204116019981263-as-1204237713388861

## Does this pull request change what data or activity we track or use?



No.

## Testing instructions:






* Go to the Social Admin page, and see if the Publicize toggle is still there and still works.
* If it's off, you should not see the sharing section in the editor.
* Changing the state should also be preserved and loaded upon page refresh
